waves-integration/src/main/scala/com/wavesplatform/dex/grpc/integration/clients/combined/CombinedStream.scala

```diff
@@ -67,11 +67,16 @@ class CombinedStream(
 
   private val mergedEvents = ConcurrentSubject.publish[Either[SystemEvent, SystemEvent]]
 
+  @volatile var blockchainStatus: Status = Status.Starting()
+
+  def status(): Status = blockchainStatus
+
   val lastStatus = mergedEvents
     .foldLeft[Status](Status.Starting()) {
       case (orig, Left(evt)) => utxEventsTransitions(orig, evt).tap(updated => log.info(s"utx: $orig + $evt -> $updated"))
       case (orig, Right(evt)) => blockchainEventsTransitions(orig, evt).tap(updated => log.info(s"bu: $orig + $evt -> $updated"))
     }
+    .tap(_.map(blockchainStatus = _))
```

Comment:
Looks like this `tap` doesn't relate to `Observable`, let's replace it by `doOnNext`

Reply:
fixed, but that doesn't work anyway

Reply:
was moved inside foldLeft, it works